diff -ruNpw SOURCE/compiler/scstate.c SOURCE_patched/compiler/scstate.c
--- SOURCE/COMPILER/scstate.c	2006-06-10 17:29:42.000000000 +0700
+++ SOURCE_patched/COMPILER/scstate.c	2012-06-18 23:18:46.751958900 +0700
@@ -185,7 +185,7 @@ SC_FUNC void state_buildlist(int **list,
   } /* if */
 
   /* find the insertion point (the list has to stay sorted) */
-  for (idx=0; idx<*count && *list[idx]<stateid; idx++)
+  for (idx=0; idx<*count && (*list)[idx]<stateid; idx++)
     /* nothing */;
   if (idx<*count)
     memmove(&(*list)[idx+1],&(*list)[idx],(int)((*count-idx+1)*sizeof(int)));
