diff -ruNpw SOURCE/COMPILER/sc.h SOURCE_patched/COMPILER/sc.h
--- SOURCE/COMPILER/sc.h	Mon Nov  6 08:54:06 2006
+++ SOURCE_patched/COMPILER/sc.h	Sun Jan 22 19:56:15 2012
@@ -51,7 +51,7 @@
 #define sCHARBITS   8       /* size of a packed character */
 
 #define sDIMEN_MAX     3    /* maximum number of array dimensions */
-#define sLINEMAX     511    /* input line length (in characters) */
+#define sLINEMAX    4095    /* input line length (in characters) */
 #define sCOMP_STACK   32    /* maximum nesting of #if .. #endif sections */
 #define sDEF_LITMAX  500    /* initial size of the literal pool, in "cells" */
 #define sDEF_AMXSTACK 4096  /* default stack size for AMX files */
